scene/gui/text_edit.cpp

```diff
@@ -4462,6 +4462,15 @@ void TextEdit::set_caret_blink_interval(const float p_interval) {
 	caret_blink_timer->set_wait_time(p_interval);
 }
 
+void TextEdit::set_draw_caret_when_readonly_enabled(bool p_value) {
+	caret_on_readonly = p_value;
+	queue_redraw();
```

Comment:
Should do this to prevent unneeded redraws.
```
   if (caret_on_readonly == p_value) {
       return;
   }
   caret_on_readonly = p_value;
   queue_redraw();
```

Reply:
Oh... yeah, I didn't understand what you meant before. I'll fix it all!